src/main/java/org/reactome/release/cosmicupdate/COSMICUpdateUtil.java

```diff
@@ -0,0 +1,320 @@
+			}
+			i++;
+		}
+		return foundMismatchedRefSequence;
+	}
+
+	/**
+	 * Gets COSMIC identifiers from the database.
+	 * Queries the database for a ReferenceDatabase named "COSMIC" and then gets all DatabaseIdentifier objects
+	 * that refer to the COSMIC ReferenceDatabase via the referenceDatabase attribute.
+	 * This method will terminate the execution of the program if more than 1 "COSMIC" ReferenceDatabase is found.
+	 * If you plan to add more "COSMIC" ReferenceDatabase objects, this code will need to be changed to use the <em>correct</em> "COSMIC"
+	 * ReferenceDatabase.
+	 * @param adaptor
+	 * @return A Collection of DatabaseIdentifier objects.
+	 * @throws SQLException
+	 * @throws Exception
+	 * @throws InvalidAttributeException
+	 */
+	static Collection<GKInstance> getCOSMICIdentifiers(MySQLAdaptor adaptor) throws SQLException, Exception, InvalidAttributeException
+	{
+		@SuppressWarnings("unchecked")
+		Collection<GKInstance> refDBs = adaptor.fetchInstanceByAttribute(ReactomeJavaConstants.ReferenceDatabase, ReactomeJavaConstants.name, " = ", "COSMIC");
+		GKInstance cosmicRefDB = null;
+		if (refDBs.size() == 1)
+		{
+			cosmicRefDB = refDBs.stream().findFirst().get();
+		}
+		else
+		{
+			logger.error("Wrong number of COSMIC refDBs: {} ; only 1 was expected.", refDBs.size());
+			logger.error("Cannot proceeed!");
+			System.exit(1);
+		}
+		
+		@SuppressWarnings("unchecked")
+		Collection<GKInstance> cosmicObjects = adaptor.fetchInstanceByAttribute(ReactomeJavaConstants.DatabaseIdentifier, ReactomeJavaConstants.referenceDatabase, " = ", cosmicRefDB.getAttributeValue(ReactomeJavaConstants.DB_ID));
+		return cosmicObjects;
+	}
+	
+	
+	/**
+	 * Produces a report on identifiers. Report indicates old/"legacy" identifiers, suggested prefixes, new identifiers suggested from COSMIC files,
+	 * and validity of old identifiers.
+	 * @param updaters The map of identifier updaters.
+	 * @throws IOException
+	 */
+	public static void printIdentifierUpdateReport(Map<String, COSMICIdentifierUpdater> updaters) throws IOException
+	{
+		// Create the reports directory if it's missing.
+		if (!Files.exists(Paths.get(COSMICUpdateUtil.reportsDirectoryPath)))
+		{
+			Files.createDirectories(Paths.get(COSMICUpdateUtil.reportsDirectoryPath));
+		}
+		try(CSVPrinter printer = new CSVPrinter(new FileWriter(COSMICUpdateUtil.reportsDirectoryPath + "/COSMIC-identifiers-report_"+dateSuffix+".csv"), CSVFormat.DEFAULT.withHeader("DB_ID", "Identifier", "Suggested Prefix", "Valid (according to COSMIC files)?", "COSV identifier", "Mutation IDs", "COSMIC Search URL")))
+		{
+			for (COSMICIdentifierUpdater record : updaters.values().parallelStream().sorted().collect(Collectors.toList()))
+			{
+				// Include a COSMIC Search URL for the identifier in the report, to make it easier for Curators to follow up on identifiers that might need attention.
+				String url;
+				String identifierForUrl = "";
+				if (record.getIdentifier().startsWith("C"))
```

Comment:
Same as https://github.com/reactome/release-cosmic-update/pull/1/files#r552224756 - though there isn't any commenting about filtering by numeric identifiers so its less of an issue

Reply:
Checking that the first character is numeric would require a regex (or parsing a string to numeric) vs. checking that the first character is "C". Checking the first character of a string is much simpler than evaluating a regular expression or parsing a string to a number (which could also involve regular expressions).

Reply:
Similar to here https://github.com/reactome/release-cosmic-update/pull/1/files#r568852047.

Since this kind of check occurs more than 2 times, could a static helper function be added for it?

Reply:
I think this expression is simple enough that a separate function is not needed.